src/coreclr/inc/corinfo.h

```diff
@@ -356,6 +356,31 @@ enum StructFloatFieldInfoFlags
     STRUCT_HAS_8BYTES_FIELDS_MASK = (STRUCT_FIRST_FIELD_SIZE_IS8 | STRUCT_SECOND_FIELD_SIZE_IS8),
 };
 
+// On RISC-V and LoongArch a struct with up to two non-empty fields, at least one of them floating,
+// can be passed in registers. FPStructInfo represents passing information for such structs.
+struct FPStructInfo
+{
+    struct Field
+    {
+        bool isFloating;
+        uint8_t size;
+        uint32_t offset;
+
+        bool Empty() const
+        {
+            static_assert(sizeof(*this) == sizeof(uint64_t), "sizeof(FPStructField) must be 64 bits");
+            return *(uint64_t*)this == 0;
+        }
+    };
+
+    Field fields[2];
+
+    bool IsIntegerOnly() const
+    {
+        return fields[0].Empty() && fields[1].Empty();
+    }
+};
```

Comment:
The new cases JIT without any assertions but I can't get any further without including size & offset of the passed fields so we need to change the API of Get(RiscV|LoongArch)64PassStructInRegisterFlags. The good news is only the return value needs to be changed, the algorithm for field flattening (the hard part) stays much the same.

@LuckyXu-HF @shushanhf I think LoongArch will also need it to pass e.g. struct { struct{}; byte; struct{}; float; } properly so I'd like to GetLoongArch64PassStructInRegisterFlags to return the same structure because there's so much shared code now.

P.s. LoongArch has the field offsets bounded to 15 so if size is of much concern to you (currently the FPStructInfo is 2 machine words), I can make the LA version of FPStructInfo smaller with bitfields so the compiler can e.g. fit values for comparison into 16-bit immediates of some LA instructions.